c10/core/Device.h

```diff
@@ -182,6 +186,14 @@ struct C10_API Device final {
         !is_cpu() || index_ <= 0,
         "CPU device index must be -1 or zero, got ",
         static_cast<int>(index_));
+#ifdef USE_CUDA
+#ifdef FBCODE_CAFFE2
+    TORCH_INTERNAL_ASSERT_DEBUG_ONLY(
+        !is_cuda() || index_ <= C10_COMPILE_TIME_MAX_GPUS,
+        "CUDA device index must be not exceeding ",
+        C10_COMPILE_TIME_MAX_GPUS);
+#endif
+#endif
```

Comment:
Can we skip this extra code to avoid adding cuda dependency here? This is already checked in other places.

Reply:
Can Skip that.

Reply:
Even if you ifdef it, it is still a cuda dependency haha
tbh if this is only for fbcode, I would remove this code altogether.